### ISSUE TITLE
MKL Pardiso error during symbolic factorization -3. (reordering_phase) error_num= -180

### ISSUE THREAD

**Issue by zearom32** [oxfordcontrol/osqp-python#14]
Hi, I'm trying to use pardiso but there is a coredump. The demo is simple:

```
P = sparse.csc_matrix([[ 3.,  2.],
     [ 2.,  3.]]
     )

q = np.array([1.0, 1.0])
A = sparse.csc_matrix([[1.0, 0.0], [0.0, 1.0]])

l = np.array([0.0, 0.0])
u = np.array([100.0,100.0])

qp = osqp.OSQP()
qp.setup(P,q,A,l,u,linsys_solver="mkl pardiso")
qp.solve()
```

The error message is 

> ERROR in init_linsys_solver_pardiso: Error during symbolic factorization: -3

Actually when input data is changed, there can be different error messages, the error code can be -2,-3.

I'm trying to figure out what happened but failed. What I find is:

- the default solver is OK

- c interface with the same data is OK

- this function https://github.com/oxfordcontrol/osqp/blob/2846ab19e3ecb2b61eee19559c85761ffaf31eb6/lin_sys/direct/pardiso/pardiso_interface.c#L51 generate a core dump

- input to the function seems good

**Comment by rwolst** [comment]
I am having a similar issue when using the `osqp` from `pypi`. If I use `conda` then I end up getting a seg fault. More details can be found here: https://github.com/oxfordcontrol/osqp/issues/65.

**Comment by bstellato** [comment]
The coredump should be fixed in the `develop` branch. However, there is an issue with using the anaconda mkl distribution. I have no problems on my mac with anaconda and numpy not linked to mkl

```
In [2]: numpy.show_config()
blas_mkl_info:
  NOT AVAILABLE
...
lapack_mkl_info:
  NOT AVAILABLE
```

However, when I try on another system where ` numpy.show_config()` returns something different than `NOT AVAILABLE` I get the symbolic factorization error -3 as shown in https://github.com/oxfordcontrol/osqp/issues/65

In particular, if I enable the output from pardiso with [msglvl](https://github.com/oxfordcontrol/osqp/blob/50b77d438066b68b2ebd18a526bf55d88d95ddb3/lin_sys/direct/pardiso/pardiso_interface.c#L161) = 1, I get the output

```
*** Error in PARDISO  (        reordering_phase) error_num= -180
*** error PARDISO: reordering, symbolic factorization
```

It seems related to some linking issue since the error appears also [here](https://software.intel.com/en-us/forums/intel-math-kernel-library/topic/292179). However, in the C library and in other interfaces I do not see it happening. I strongly suspect it is related to how anaconda manages the python/numpy mkl linkage.

**Comment by bstellato** [comment]
Possiibly related issues:

https://github.com/JuliaPy/PyCall.jl/issues/443
https://github.com/ContinuumIO/anaconda-issues/issues/3823
https://github.com/numpy/numpy/issues/5906

**Comment by bstellato** [comment]
Other very related issue. Other people getting the same exact problem in Julia: https://github.com/JuliaSparse/Pardiso.jl/issues/28

**Comment by bstellato** [comment]
This is now fixed in the `develop` branch in 60f307bd8aa2c2fd369de46119341ea445860a33. It is related to the fact that numpy is compiled with 32bit ints support by default (see the links above).

Therefore, anaconda links numpy to the 32bit ints MKL library. If OSQP tries to use the same library with 64bit ints it creates memory issues. Too bad MKL does not detect that and reports a wrong error.

We now disabled 64bit integer support in python by default. It can be enabled with a simple flag in setup.py.
